xbmc/playlists/PlayListXSPF.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ *  Copyright (C) 2018 Tyler Szabo
+ *  Copyright (C) 2018 Team Kodi
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "PlayListXSPF.h"
+
+#include "utils/log.h"
+#include "utils/URIUtils.h"
+#include "utils/XBMCTinyXML.h"
+#include "URL.h"
+
+using namespace PLAYLIST;
+
+static constexpr char const* LOCATION_TAGNAME = "location";
+static constexpr char const* PLAYLIST_TAGNAME = "playlist";
+static constexpr char const* TITLE_TAGNAME = "title";
+static constexpr char const* TRACK_TAGNAME = "track";
+static constexpr char const* TRACKLIST_TAGNAME = "trackList";
+
+CPlayListXSPF::CPlayListXSPF(void) = default;
+
+CPlayListXSPF::~CPlayListXSPF(void) = default;
+
+static inline const std::string GetXMLText(const TiXmlElement* const pXmlElement)
```

Comment:
Sidenote: You can remove the `inline`, it doesn't actually do anything here

Also why is the return value `const`?

Reply:
Good point! Fixed, thanks :)

I initially wrote this as `const std::string&` out of habit (got rid of the reference because of a [C++11 nuance](https://stackoverflow.com/questions/10231349/are-the-days-of-passing-const-stdstring-as-a-parameter-over)).

Reply:
Good catch. I initially was returning `const std::string&` due to a auto-pilot thinking of it as an accessor except that made no sense because it existed inside the function's stack.